src/llama_stack_provider_kft/kft_adapter.py

```diff
@@ -52,6 +54,87 @@ async def supervised_fine_tune(
     ) -> PostTrainingJob:
         if isinstance(algorithm_config, LoraFinetuningConfig):
             raise NotImplementedError()
+        
+        if self.config.preprocess:
+            # create a pod and just run DP on it
+            # output data to PVC dir
+            # delete pod so PVC can be reused
+            # Define the multi-line command script
+            # oc policy add-role-to-user edit system:serviceaccount:default:my-service-account
+            import kubernetes
+            kubernetes.config.load_kube_config()  # Use ServiceAccount credentials
+
+            # take local SDG data on disk
+            # validate k8s connection
+            # create PVC
+            # upload data to PVC
+
+            # Request the Kubernetes API
+            v1 = kubernetes.client.CoreV1Api()
+            dp_master_args = f"""
+            echo "Processing data"
+            echo "Using {self.config.model_path} model for training"
+            echo "Using {self.config.data_path} data for training"
+
+            pip install instructlab-training
+
+            python -c '
+            import instructlab.training.data_process as dp
+            from instructlab.training import (
+                DataProcessArgs,
+                TrainingArgs,
+            )
+
+            dp.main(
+                DataProcessArgs(
+                    data_output_path={os.path.dirname(self.config.data_path)},
+                    model_path={self.config.model_path},
+                    data_path={self.config.data_path},
+                    max_seq_len={training_config.max_seq_len},
+                    chat_tmpl_path={self.config.chat_tmpl_path},
+                )
+            )
+            '
+            """
+
+            pod_manifest = {
+                "apiVersion": "v1",
+                "kind": "Pod",
+                "metadata": {
+                    "name": "ilab-data-processor",
+                },
+                "spec": {
+                    "containers": [
```

Comment:
I wonder could it be run as an init container of the PyTorchJob so creating and watching a separate pod is not needed?

Reply:
+1 on the init container for the PyTorchJob master's pod.

Reply:
cool, will switch to this